llvm/lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -707,6 +707,10 @@ void TargetPassConfig::addPass(Pass *P) {
   // and shouldn't reference it.
   AnalysisID PassID = P->getPassID();
 
+  IdentifyingPassPtr TargetID = getPassSubstitution(PassID);
+  if (!overridePass(PassID, TargetID).isValid())
+    return;
```

Comment:
Shouldn't need to touch the override infrastructure?

Reply:
Sorry, I don't get it... What do you mean? 

I had to add this to check if the pass was disabled. It's a copy-and-paste from the `addPass(AnalysisID PassID)` method.

Reply:
Yes, that's the problem. You shouldn't have to do anything like this

Reply:
I've replaced this check with a `isPassSubstitutedOrOverridden` check, similar to how it's done for `createPrologEpilogInserterPass`. Does this address your comment?